extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -1470,6 +1493,28 @@ public static Table readKeyValuePartitionedTable(
                 readInstructions, tableDefinition);
     }
 
+    /**
+     * Creates a partitioned table via the key-value partitioned parquet files from the root {@code directory} using the
+     * provided {@code tableDefinition}.
+     *
+     * @param directory the path or URI for the root directory to search for .parquet files
+     * @param readInstructions the instructions for customizations while reading
+     * @param tableDefinition the table definition
+     * @return the table
+     * @see ParquetKeyValuePartitionedLayout#ParquetKeyValuePartitionedLayout(URI, TableDefinition, ParquetInstructions)
+     * @see #readPartitionedTable(TableLocationKeyFinder, ParquetInstructions, TableDefinition)
+     */
+    public static Table readKeyValuePartitionedTable(
+            @NotNull final String directory,
+            @NotNull final ParquetInstructions readInstructions,
+            @NotNull final TableDefinition tableDefinition) {
```

Comment:
You proposed moving the `TableDefinition` into `ParquetInstructions`. It's a good idea. I also think we should stop making `File` overloads for new functions.
We could avoid breaking existing overloads... allow `TableDefinition` to be passed, but then the instructions should have an equal definition or no definition.
You could also enumerate known layouts.

Reply:
@devinrsmith What do you think of this proposal?
That way we standardize on `readTable(String)` and `readTable(String, ParquetInstructions)`. Existing extraneous overloads can stay, `@Deprecated`, just not be used in our documentation.